### ISSUE TITLE
[book] Document the standard library

### ISSUE THREAD

**Issue by vanillajonathan** [PRQL/prql#1761]
https://prql-lang.org/book/language-features/standard-library.html

The standard library page in the documentation just includes the source code for `std.prql`.

I would like fuller documentation; a paragraph for each function that describes what it does.

**Comment by max-sixty** [comment]
Agree. This would make a nice contribution. Especially since we added types to the stdlib, it's not really in a user-presentable form.

Even if we just had a table of functions and their SQL equivalent, that would be much nicer...

**Comment by vanillajonathan** [comment]
Yes, I am unable to read and understand the `std.prql` file.

I see some stuff like `from<table>` and I was thinking those are generics? But I have a feeling they're not. I am confused.

**Comment by max-sixty** [comment]
The important parts of the stdlib to explain are the functions — `min` / `max` etc. So if someone wanted to make a table of those, that would be welcome. The `foo<table>` is a type annotation; that's not needed to make a contribution here.

**Comment by vanillajonathan** [comment]
This type annotation syntax is very foreign to me, and I can imagine to others as well. In my head that syntax is for generics.

**Comment by aljazerzen** [comment]
The long-term solution would be to introduce "structured comments" to the language. Something like `///` in Rust or `/**` in Java. These comments could then be extracted and converted to HTML which would be inserted into the book (or be a standalone site), similar to what rustdoc are doing javadoc.

I don't think we have the manpower to do this now, but as a start, we can document stdlib *in stdlib.prql*:

```elm
## Finds minimum value of a column
##
## Example:
## ```
## from tracks
## select (min milliseconds)
## ```
func min <scalar || column> column -> null
```
